### PR TITLE
TypeScript support.

### PR DESCRIPTION
**Pull request by marcogrcr**
*Issue #, if available:* #15

*Description of changes:*

WHAT?

- Added `src/index.d.ts` typings for usage with TypeScript.
- Reworded the API documentation to make it more clearer.

WHY?

Add TypeScript compatibility.

I faced several issues trying to use `amazon-connect-chatjs` due to
unclear documentation. I have reworded the API documentation to
make it clearer for customers.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,143 +71,326 @@ $ git clone https://github.com/amazon-connect/amazon-connect-chatjs
...
+
+The arguments are based on the [API request body](https://docs.aws.amazon.com/connect-participant/latest/APIReference/API_SendEvent.html#API_SendEvent_RequestSyntax) with the following differences:
+
+- Fields are in `camelCase`.
+- `ClientToken` cannot be specified.
+- `ContentType` allows the following values:
+  - `"application/vnd.amazonaws.connect.event.chat.ended"`
+  - `"application/vnd.amazonaws.connect.event.participant.joined"`
+  - `"application/vnd.amazonaws.connect.event.participant.left"`
+  - `"application/vnd.amazonaws.connect.event.transfer.succeeded"`
```
Comment (marcogrcr) [inline README.md:273]: The previous documentation incorrectly referred to this event as `application/vnd.amazonaws.connect.event.transfer.succeed` (notice `succeeded` vs `succeed`).

**Comment by silbinarywolf** [comment]
@ctwomblyamzn or @johnmryan, can either of you please take some time to resolve these package.json merge conflicts and merge this in before too much time passes?

**Comment by Christian-Sesta** [comment]
@marcogrcr would you be able to resolve these merge conflicts so this can be merged? @ctwomblyamzn have there been any big changes to the core types since this PR was posted? Might be worth making sure nothing new was missed?

**Comment by vertic4l** [comment]
ChatJS definitely needs some more TypeScript Support. @marcogrcr can you go on?

**Comment by marcogrcr** [comment]
@Christian-Sesta and @vertic4l : Sorry it took so long, I was swamped with other stuff. Now it's up to the package maintainers (e.g. @ctwomblyamzn) to accept and merge.

**Comment by Christian-Sesta** [comment]
@johnmryan would you be able to review and/ or merge this before to much time passes. Would be great to prevent this having merge conflict again and taking another few months to actually get looked at.

**Review by vertic4l — APPROVED** [review]

**Review by johnmryan — APPROVED** [review]
Looks good to me, thank you for taking the time to add types and document all of this, @marcogrcr .

Apologies for the delay in reviewing this code, we will stay more on top of PRs in the future.

**Pull request merged**